core/src/com/unciv/ui/cityscreen/CityConstructionsTable.kt

```diff
@@ -432,6 +435,43 @@ class CityConstructionsTable(val cityScreen: CityScreen) : Table(CameraStageBase
 
         return button
     }
+    
+    private fun getBuyWithFaithButton(construction: IConstruction?): TextButton? {
```

Comment:
So much here is common with the getButButton that I think it can be changed to 'getBuyButton(construction, withFaith:Boolean=false)